andela/ah-backend-odin
# #161340179 Rate articles

#### What does this PR do?
Ensures that users of authors haven can rate articles posted by others
#### Description of Task to be completed?
Once there exists an article, the user should be able to rate this article for the range of 1 to 5
#### How should this be manually tested?
Assuming an article is already created and the server is running, 
1. Go to go to your postman and login to the system,
2. Get the auth token and put it in the Authorization as bearer token. 
3. Test posting rating using the url `http://127.0.0.1:8000/api/articles/<slug>/ratings/`, and 
4. Provide a rate of ur choice in json format as { "article_rate" : 5} and post.
  For testing purpose, you can login using another user account and rate the same article in order to be able to see average rate.
5. To see the average rating of a particular article, hit a get on postman through this link `http://127.0.0.1:8000/api/articles/<slug>/viewratings/`.
6. To see average ratings of all articles respectively,  hit a get on postman through this link `http://127.0.0.1:8000/api/articles/ratings/`.
7. NB: the `<slug>` in the urls provided above needs to be replaced by the actual article under test.
#### Any background context you want to provide?
1. Delete and edit rates were not implemented as they are not a good design criteria for rating anything
2. Validations covered: Article author can't rate his own article, users can't rate the same article more than once, and rates are between 1 to 5 only.
#### What are the relevant pivotal tracker stories?
[#161340179](https://www.pivotaltracker.com/story/show/161340179)
#### Screenshots (if appropriate)
1. Posting a rating of an article
<img width="972" alt="post rate" src="https://user-images.githubusercontent.com/20398831/48292150-4770e180-e48a-11e8-9030-1f9b81b5cf31.png">

2. Viewing rating of one article


<img width="968" alt="get one rate" src="https://user-images.githubusercontent.com/20398831/48292355-0d540f80-e48b-11e8-9ad8-0c9c25fed65e.png">

3. Getting All Articles With Their Rates
<img width="955" alt="all articles" src="https://user-images.githubusercontent.com/20398831/48292502-99fecd80-e48b-11e8-8302-ed909dd9137b.png">
